pkg/admission/apibinding/apibinding_admission.go

```diff
@@ -0,0 +1,225 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apibinding
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+
+	kcpadmissionhelpers "github.com/kcp-dev/kcp/pkg/admission/helpers"
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	"github.com/kcp-dev/kcp/pkg/apis/tenancy/v1alpha1/helper"
+)
+
+const (
+	PluginName = "apis.kcp.dev/APIBinding"
+)
+
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName,
+		func(_ io.Reader) (admission.Interface, error) {
+			return &apiBindingAdmission{
+				Handler:          admission.NewHandler(admission.Create, admission.Update),
+				createAuthorizer: kcpadmissionhelpers.NewAdmissionAuthorizer,
+			}, nil
+		})
+}
+
+type apiBindingAdmission struct {
+	*admission.Handler
+	kubeClusterClient *kubernetes.Cluster
+
+	createAuthorizer kcpadmissionhelpers.AdmissionAuthorizerFactory
+}
+
+// Ensure that the required admission interfaces are implemented.
+var _ = admission.ValidationInterface(&apiBindingAdmission{})
+var _ = admission.MutationInterface(&apiBindingAdmission{})
+var _ = admission.InitializationValidator(&apiBindingAdmission{})
+
+// Validate validates the creation and updating of APIBinding resources. It also performs a SubjectAccessReview
+// making sure the user is allowed to use the 'bind' verb with the referenced APIExport.
+func (o *apiBindingAdmission) Validate(ctx context.Context, a admission.Attributes, _ admission.ObjectInterfaces) error {
+	if a.GetResource().GroupResource() != apisv1alpha1.Resource("apibindings") {
+		return nil
+	}
+
+	obj, err := kcpadmissionhelpers.NativeObject(a.GetObject())
```

Comment:
with https://github.com/kcp-dev/kcp/pull/577 merged this will not compile anymore

Reply:
Fixed